### ISSUE TITLE
Could not find libcdada - github master

### ISSUE THREAD

**Issue by rodecker** [pmacct/pmacct#623]
**Description**
I am attempting to build an rpm from recent github sources, but am running into the exact issue described in #501.
The solution mentioned in this issue is downloading the latest official tar.gz. From this I can indeed build the binaries. However, this is not an option for me, as I would like to include changes/fixes that have been made latter (it has been quite a while since the latest release).

Could you provide some guidance on how to resolve this?

**Version**
github master (2022-07-14)

**Comment by paololucente** [comment]
Hi Martin ( @rodecker ),

In the issue you reference the problem - long story short - is using the artifact generated by GitHub; we include our own artifact and that one should be preferred. I think you should not be hitting any issue just picking a specific commit (or latest master) and compiling it and building a distribution package for it.

So, one step back: are you generally able to compile pmacct? Do you have some terminal log for us to look into?

Paolo

**Comment by rodecker** [comment]
Hi Paolo,

I found my problem. The official tar.gz includes libcdada in the tarball, while a `git clone` does not (the library is downloaded during the `make` step). My build spec did not contain git, causing the check in https://github.com/pmacct/pmacct/blob/master/configure.ac#L1186 to silently fail and set `WITH_EXTERNAL_DEPS_DEFAULT=no`. This then lead to the `Could not find libcdada` error in the `configure` step.

I resolved this by using `git clone --recurse-submodules https://github.com/pmacct/pmacct.git` to download the dependency before building. I was then able to execute the build successfully.

Cheers,
Martin
